v2/pink-sb/src/lib/sidebar/Base.svelte

```diff
@@ -0,0 +1,192 @@
+<script lang="ts">
+    import { Badge, Icon } from '$lib/index.js';
+    import { IconChevronLeft, IconChevronRight, IconMinus } from '@appwrite.io/pink-icons-svelte';
+    import type { $$Props } from '$lib/sidebar/index.js';
+
+    export let state: $$Props['state'] = 'open';
+    export let resizable: $$Props['resizable'] = true;
+</script>
+
+{#if resizable}
+    <button
+        class="collapse"
+        class:icons={state === 'icons'}
+        on:click={() => {
+            state = state === 'icons' ? 'open' : 'icons';
+        }}
+    >
+        <div class="lines-container">
+            <div class="icon icon-idle minus-icon"><Icon icon={IconMinus} /></div>
+            <div class="icon icon-hover">
+                <Icon icon={state === 'icons' ? IconChevronRight : IconChevronLeft} />
+            </div>
+        </div>
+        <div class="badge">
+            <Badge
+                size="xs"
+                variant="primary"
+                content={state === 'icons' ? 'Expand' : 'Collapse'}
+            />
+        </div>
+    </button>
+{/if}
+<nav
+    class:only-icons={state === 'icons'}
+    class:open={state === 'open'}
+    class:closed={state === 'closed'}
+    {...$$props}
+>
+    <slot name="top"></slot>
+    <div class="middle">
+        <slot name="middle"></slot>
+    </div>
+    <slot name="bottom"></slot>
+</nav>
+
+<style lang="scss">
+    nav {
+        position: fixed;
```

Comment:
I think this position fixed might not work in a complete layout, can you try to add the heading and some content in the Story so we can check?